diff --git a/lib/directives/ripple/index.mjs b/lib/directives/ripple/index.mjs
index a060ba93fb9784ae9bd4c4ac4cd3dcf57e4a76f8..7ba99f2d907e7520fac48d286ea22a38b4c7d1b0 100644
--- a/lib/directives/ripple/index.mjs
+++ b/lib/directives/ripple/index.mjs
@@ -83,11 +83,13 @@ const ripples = {
     animation.classList.add('v-ripple__animation--visible');
     transform(animation, `translate(${x}, ${y}) scale3d(${scale},${scale},${scale})`);
     animation.dataset.activated = String(performance.now());
-    setTimeout(() => {
-      animation.classList.remove('v-ripple__animation--enter');
-      animation.classList.add('v-ripple__animation--in');
-      transform(animation, `translate(${centerX}, ${centerY}) scale3d(1,1,1)`);
-    }, 0);
+    requestAnimationFrame(() => {
+      requestAnimationFrame(() => {
+        animation.classList.remove('v-ripple__animation--enter');
+        animation.classList.add('v-ripple__animation--in');
+        transform(animation, `translate(${centerX}, ${centerY}) scale3d(1,1,1)`);
+      })
+    })
   },
   hide(el) {
     if (!el?._ripple?.enabled) return;
diff --git a/lib/styles/tools/_states.sass b/lib/styles/tools/_states.sass
index 68558cac9764018d80ec6a4b2ee90b741ea35104..79ebddbeed07b81cce6eac6a51b6069b30e55c08 100644
--- a/lib/styles/tools/_states.sass
+++ b/lib/styles/tools/_states.sass
@@ -6,37 +6,31 @@
   @if string.slice(string.unquote($selector), 1, 1) != '&'
     $selector: #{'>'} #{$selector}
 
-  &:hover
-    #{$selector}
-      opacity: calc(#{map.get(settings.$states, 'hover')} * var(--v-theme-overlay-multiplier))
-
-  &:focus-visible
-    #{$selector}
-      opacity: calc(#{map.get(settings.$states, 'focus')} * var(--v-theme-overlay-multiplier))
+  @media (hover: hover)
+    &:hover
+      #{$selector}
+        opacity: calc(#{map.get(settings.$states, 'hover')} * var(--v-theme-overlay-multiplier))
 
-  @supports not selector(:focus-visible)
-    &:focus
+  @supports selector(:focus-visible)
+    &:focus-visible
       #{$selector}
         opacity: calc(#{map.get(settings.$states, 'focus')} * var(--v-theme-overlay-multiplier))
 
   @if ($active)
-    &--active,
-    &[aria-haspopup="menu"][aria-expanded="true"]
+    &--active
       @include active-states($selector)
 
 @mixin active-states ($selector, $base: map.get(settings.$states, 'activated'))
   #{$selector}
     opacity: calc(#{$base} * var(--v-theme-overlay-multiplier))
 
-  &:hover
-    #{$selector}
-      opacity: calc((#{$base} + #{map.get(settings.$states, 'hover')}) * var(--v-theme-overlay-multiplier))
-
-  &:focus-visible
-    #{$selector}
-      opacity: calc((#{$base} + #{map.get(settings.$states, 'focus')}) * var(--v-theme-overlay-multiplier))
+  @media (hover: hover)
+    &:hover
+      #{$selector}
+        opacity: calc((#{$base} + #{map.get(settings.$states, 'hover')}) * var(--v-theme-overlay-multiplier))
 
-  @supports not selector(:focus-visible)
-    &:focus
+  @supports selector(:focus-visible)
+    &:focus-visible
       #{$selector}
         opacity: calc((#{$base} + #{map.get(settings.$states, 'focus')}) * var(--v-theme-overlay-multiplier))
+
